### PR TITLE
add promise.new

### PR DESCRIPTION
**Pull request by CrowdHailer**

### CODE REVIEW

**Comment by lpil** [comment]
Hello! What's the use-case for this function?

**Comment by CrowdHailer** [comment]
Today, I have bindings to setTimeout that I would like to convert to promises.

**Comment by lpil** [comment]
Is this to push something onto the next tick? I think it'd be best to use the existing `await` function rather than add more to this library.

**Comment by CrowdHailer** [comment]
> Is this to push something onto the next tick? 
Don't know what you mean here. 

But I don't see how I can use the existing await. because I don't have an initial promise to start with.

File: src/gleam/javascript/promise.gleam [hunk 1]
```diff
@@ -5,6 +5,9 @@ import gleam/javascript/array.{Array}
 // TODO: labels
 pub type Promise(value)
 
+@external(javascript, "../../ffi.mjs", "newPromise")
+pub fn new(a: fn(fn(value) -> Nil) -> Nil) -> Promise(value)
```
Comment (lpil) [inline src/gleam/javascript/promise.gleam:9]: Could we have some tests for this please? We'll need to include a test to make sure that nested promises don't collapse.

Comment (CrowdHailer) [reply src/gleam/javascript/promise.gleam:9]: done

**Review by lpil — APPROVED** [review]
Thank you

**Pull request merged**